### ISSUE TITLE
"Unrecognized input header" while trying to access logs

### ISSUE THREAD

**Issue by voxik** [kubernetes/kubernetes#25194]
I observe the following error:

```
$ kubectl logs my-worker-6698i --v=10

... snip ...

I0505 08:10:11.075361    7746 round_trippers.go:267] curl -k -v -XGET  -H "User-Agent: kubectl/v1.2.0 (linux/amd64) kubernetes/738b760" -H "Accept: application/json, */*" http://localhost:8080/apis
I0505 08:10:11.075620    7746 round_trippers.go:286] GET http://localhost:8080/apis 200 OK in 0 milliseconds
I0505 08:10:11.075646    7746 round_trippers.go:292] Response Headers:
I0505 08:10:11.075652    7746 round_trippers.go:295]     Content-Type: application/json
I0505 08:10:11.075656    7746 round_trippers.go:295]     Date: Thu, 05 May 2016 12:10:11 GMT
I0505 08:10:11.075660    7746 round_trippers.go:295]     Content-Length: 18
I0505 08:10:11.075676    7746 request.go:870] Response Body: {
  "groups": []
}
I0505 08:10:11.075840    7746 round_trippers.go:267] curl -k -v -XGET  -H "User-Agent: kubectl/v1.2.0 (linux/amd64) kubernetes/738b760" http://localhost:8080/api/v1/namespaces/default/pods/my-worker-6698i/log
I0505 08:10:11.090244    7746 round_trippers.go:286] GET http://localhost:8080/api/v1/namespaces/default/pods/my-worker-6698i/log 500 Internal Server Error in 14 milliseconds
I0505 08:10:11.090268    7746 round_trippers.go:292] Response Headers:
I0505 08:10:11.090274    7746 round_trippers.go:295]     Content-Type: application/json
I0505 08:10:11.090279    7746 round_trippers.go:295]     Date: Thu, 05 May 2016 12:10:11 GMT
I0505 08:10:11.090284    7746 round_trippers.go:295]     Content-Length: 231
I0505 08:10:11.090645    7746 helpers.go:172] server response object: [{
  "metadata": {},
  "status": "Failure",
  "message": "Internal error occurred: Unrecognized input header",
  "reason": "InternalError",
  "details": {
    "causes": [
      {
        "message": "Unrecognized input header"
      }
    ]
  },
  "code": 500
}]
F0505 08:10:11.090668    7746 helpers.go:107] Error from server: Internal error occurred: Unrecognized input header
```

While accessing the logs via docker works just fine:

```
$ docker logs k8s_my-worker.f4afcc6d_my-worker-6698i_default_1d8c1025-12b7-11e6-a7f7-5254004a5b75_a0d48606
```

Using the following kubernetes:

```
$ rpm -q kubernetes
kubernetes-1.2.0-0.6.alpha1.git8632732.el7.x86_64
```

coming from projectatomic/adb Vagrant box.

[1] https://atlas.hashicorp.com/projectatomic/boxes/adb/

**Comment by ncdc** [comment]
Are you running Docker 1.10?

**Comment by bgrant0607** [comment]
See also #16670

**Comment by ncdc** [comment]
@bgrant0607 this has been fixed in master and 1.2.0+. This particular Kube above is one that we build from OpenShift's Godep'd copy of Kube. It does not contain the fix to fsouza/go-dockerclient.

**Comment by bgrant0607** [comment]
Thanks, @ncdc. Is there a reason this is still open?

**Comment by ncdc** [comment]
I would say it's safe to close and that there should be a big reported
elsewhere for this particular issue with the vagrant box.

**Comment by shenshouer** [comment]
kubernetes version 1.3.2 logs pod issue:

```
core@worker-x-x-x-x ~ $ kubectl get po
NAME                     READY     STATUS    RESTARTS   AGE
lfec924f1f2c9cad-4502j   1/1       Running   0          52m
lfec924f1f2c9cad-4n6ve   1/1       Running   0          52m
lfec924f1f2c9cad-5andd   1/1       Running   0          52m
lfec924f1f2c9cad-9fniy   1/1       Running   0          52m
lfec924f1f2c9cad-gw9io   1/1       Running   0          52m
lfec924f1f2c9cad-jgsh8   1/1       Running   0          52m
lfec924f1f2c9cad-sx0d9   1/1       Running   0          52m
lfec924f1f2c9cad-tf03u   1/1       Running   0          52m
lfec924f1f2c9cad-zersd   1/1       Running   0          52m
lfec924f1f2c9cad-zq6u7   1/1       Running   0          52m
m420a6f6f2343ac4-b897n   1/1       Running   0          1h
core@worker-x-x-x-x ~ $ kubectl logs lfec924f1f2c9cad-4502j
Error from server: Unrecognized input header


```

**Comment by shenshouer** [comment]
```
~ $ kubectl logs lfec924f1f2c9cad-4502j --v=10
I0720 17:59:09.662793   14663 loader.go:327] Config loaded from file /home/core/.kube/config
I0720 17:59:09.663252   14663 round_trippers.go:299] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: kubectl/v1.3.2 (linux/amd64) kubernetes/9bafa34" http://localhost:8080/api
I0720 17:59:09.666156   14663 round_trippers.go:318] GET http://localhost:8080/api 200 OK in 2 milliseconds
I0720 17:59:09.666176   14663 round_trippers.go:324] Response Headers:
I0720 17:59:09.666180   14663 round_trippers.go:327]     Content-Type: application/json
I0720 17:59:09.666198   14663 round_trippers.go:327]     Date: Wed, 20 Jul 2016 09:59:09 GMT
I0720 17:59:09.666201   14663 round_trippers.go:327]     Content-Length: 131
I0720 17:59:09.666254   14663 request.go:891] Response Body: {"kind":"APIVersions","versions":["v1"],"serverAddressByClientCIDRs":[{"clientCIDR":"0.0.0.0/0","serverAddress":"x-x-x.6:443"}]}
I0720 17:59:09.666521   14663 round_trippers.go:299] curl -k -v -XGET  -H "User-Agent: kubectl/v1.3.2 (linux/amd64) kubernetes/9bafa34" -H "Accept: application/json, */*" http://localhost:8080/apis
I0720 17:59:09.667058   14663 round_trippers.go:318] GET http://localhost:8080/apis 200 OK in 0 milliseconds
I0720 17:59:09.667072   14663 round_trippers.go:324] Response Headers:
I0720 17:59:09.667077   14663 round_trippers.go:327]     Content-Type: application/json
I0720 17:59:09.667095   14663 round_trippers.go:327]     Date: Wed, 20 Jul 2016 09:59:09 GMT
I0720 17:59:09.667101   14663 round_trippers.go:327]     Content-Length: 1625
I0720 17:59:09.667149   14663 request.go:891] Response Body: {"kind":"APIGroupList","groups":[{"name":"apps","versions":[{"groupVersion":"apps/v1alpha1","version":"v1alpha1"}],"preferredVersion":{"groupVersion":"apps/v1alpha1","version":"v1alpha1"},"serverAddressByClientCIDRs":[{"clientCIDR":"0.0.0.0/0","serverAddress":"x-x-x.6:443"}]},{"name":"autoscaling","versions":[{"groupVersion":"autoscaling/v1","version":"v1"}],"preferredVersion":{"groupVersion":"autoscaling/v1","version":"v1"},"serverAddressByClientCIDRs":[{"clientCIDR":"0.0.0.0/0","serverAddress":"x-x-x.6:443"}]},{"name":"batch","versions":[{"groupVersion":"batch/v1","version":"v1"},{"groupVersion":"batch/v2alpha1","version":"v2alpha1"}],"preferredVersion":{"groupVersion":"batch/v1","version":"v1"},"serverAddressByClientCIDRs":[{"clientCIDR":"0.0.0.0/0","serverAddress":"x-x-x.6:443"}]},{"name":"extensions","versions":[{"groupVersion":"extensions/v1beta1","version":"v1beta1"}],"preferredVersion":{"groupVersion":"extensions/v1beta1","version":"v1beta1"},"serverAddressByClientCIDRs":[{"clientCIDR":"0.0.0.0/0","serverAddress":"x-x-x.6:443"}]},{"name":"policy","versions":[{"groupVersion":"policy/v1alpha1","version":"v1alpha1"}],"preferredVersion":{"groupVersion":"policy/v1alpha1","version":"v1alpha1"},"serverAddressByClientCIDRs":[{"clientCIDR":"0.0.0.0/0","serverAddress":"x-x-x.6:443"}]},{"name":"rbac.authorization.k8s.io","versions":[{"groupVersion":"rbac.authorization.k8s.io/v1alpha1","version":"v1alpha1"}],"preferredVersion":{"groupVersion":"rbac.authorization.k8s.io/v1alpha1","version":"v1alpha1"},"serverAddressByClientCIDRs":[{"clientCIDR":"0.0.0.0/0","serverAddress":"x-x-x.6:443"}]}]}
I0720 17:59:09.667730   14663 round_trippers.go:299] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: kubectl/v1.3.2 (linux/amd64) kubernetes/9bafa34" http://localhost:8080/api
I0720 17:59:09.668099   14663 round_trippers.go:318] GET http://localhost:8080/api 200 OK in 0 milliseconds
I0720 17:59:09.668110   14663 round_trippers.go:324] Response Headers:
I0720 17:59:09.668113   14663 round_trippers.go:327]     Content-Type: application/json
I0720 17:59:09.668117   14663 round_trippers.go:327]     Date: Wed, 20 Jul 2016 09:59:09 GMT
I0720 17:59:09.668120   14663 round_trippers.go:327]     Content-Length: 131
I0720 17:59:09.668144   14663 request.go:891] Response Body: {"kind":"APIVersions","versions":["v1"],"serverAddressByClientCIDRs":[{"clientCIDR":"0.0.0.0/0","serverAddress":"x-x-x.6:443"}]}
I0720 17:59:09.668261   14663 round_trippers.go:299] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: kubectl/v1.3.2 (linux/amd64) kubernetes/9bafa34" http://localhost:8080/apis
I0720 17:59:09.668682   14663 round_trippers.go:318] GET http://localhost:8080/apis 200 OK in 0 milliseconds
I0720 17:59:09.668696   14663 round_trippers.go:324] Response Headers:
I0720 17:59:09.668713   14663 round_trippers.go:327]     Content-Type: application/json
I0720 17:59:09.668717   14663 round_trippers.go:327]     Date: Wed, 20 Jul 2016 09:59:09 GMT
I0720 17:59:09.668720   14663 round_trippers.go:327]     Content-Length: 1625
I0720 17:59:09.668756   14663 request.go:891] Response Body: {"kind":"APIGroupList","groups":[{"name":"apps","versions":[{"groupVersion":"apps/v1alpha1","version":"v1alpha1"}],"preferredVersion":{"groupVersion":"apps/v1alpha1","version":"v1alpha1"},"serverAddressByClientCIDRs":[{"clientCIDR":"0.0.0.0/0","serverAddress":"x-x-x.6:443"}]},{"name":"autoscaling","versions":[{"groupVersion":"autoscaling/v1","version":"v1"}],"preferredVersion":{"groupVersion":"autoscaling/v1","version":"v1"},"serverAddressByClientCIDRs":[{"clientCIDR":"0.0.0.0/0","serverAddress":"x-x-x.6:443"}]},{"name":"batch","versions":[{"groupVersion":"batch/v1","version":"v1"},{"groupVersion":"batch/v2alpha1","version":"v2alpha1"}],"preferredVersion":{"groupVersion":"batch/v1","version":"v1"},"serverAddressByClientCIDRs":[{"clientCIDR":"0.0.0.0/0","serverAddress":"x-x-x.6:443"}]},{"name":"extensions","versions":[{"groupVersion":"extensions/v1beta1","version":"v1beta1"}],"preferredVersion":{"groupVersion":"extensions/v1beta1","version":"v1beta1"},"serverAddressByClientCIDRs":[{"clientCIDR":"0.0.0.0/0","serverAddress":"x-x-x.6:443"}]},{"name":"policy","versions":[{"groupVersion":"policy/v1alpha1","version":"v1alpha1"}],"preferredVersion":{"groupVersion":"policy/v1alpha1","version":"v1alpha1"},"serverAddressByClientCIDRs":[{"clientCIDR":"0.0.0.0/0","serverAddress":"x-x-x.6:443"}]},{"name":"rbac.authorization.k8s.io","versions":[{"groupVersion":"rbac.authorization.k8s.io/v1alpha1","version":"v1alpha1"}],"preferredVersion":{"groupVersion":"rbac.authorization.k8s.io/v1alpha1","version":"v1alpha1"},"serverAddressByClientCIDRs":[{"clientCIDR":"0.0.0.0/0","serverAddress":"x-x-x.6:443"}]}]}
I0720 17:59:09.669134   14663 round_trippers.go:299] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: kubectl/v1.3.2 (linux/amd64) kubernetes/9bafa34" http://localhost:8080/api/v1/namespaces/cloud/pods/lfec924f1f2c9cad-4502j
I0720 17:59:09.670670   14663 round_trippers.go:318] GET http://localhost:8080/api/v1/namespaces/cloud/pods/lfec924f1f2c9cad-4502j 200 OK in 1 milliseconds
I0720 17:59:09.670683   14663 round_trippers.go:324] Response Headers:
I0720 17:59:09.670687   14663 round_trippers.go:327]     Content-Type: application/json
I0720 17:59:09.670690   14663 round_trippers.go:327]     Date: Wed, 20 Jul 2016 09:59:09 GMT
I0720 17:59:09.670733   14663 request.go:891] Response Body: {"kind":"Pod","apiVersion":"v1","metadata":{"name":"lfec924f1f2c9cad-4502j","generateName":"lfec924f1f2c9cad-","namespace":"cloud","selfLink":"/api/v1/namespaces/cloud/pods/lfec924f1f2c9cad-4502j","uid":"5635f89e-4e58-11e6-a158-0cc47aa48410","resourceVersion":"186679611","creationTimestamp":"2016-07-20T08:59:59Z","labels":{"app":"lfec924f1f2c9cad","deployment":"v1"},"annotations":{"kubernetes.io/created-by":"{\"kind\":\"SerializedReference\",\"apiVersion\":\"v1\",\"reference\":{\"kind\":\"ReplicationController\",\"namespace\":\"cloud\",\"name\":\"lfec924f1f2c9cad\",\"uid\":\"56606a44-4e58-11e6-bd9c-0cc47aa4cd6a\",\"apiVersion\":\"v1\",\"resourceVersion\":\"186679473\"}}\n"}},"spec":{"volumes":[{"name":"default-token-sdtzw","secret":{"secretName":"default-token-sdtzw"}}],"containers":[{"name":"lfec924f1f2c9cad","image":"dhub.yunpro.cn/cuiweiqiang/crawler:latest","ports":[{"containerPort":80,"protocol":"TCP"}],"resources":{},"volumeMounts":[{"name":"default-token-sdtzw","readOnly":true,"mountPath":"/var/run/secrets/kubernetes.io/serviceaccount"}],"lifecycle":{},"terminationMessagePath":"/dev/termination-log","imagePullPolicy":"Always","stdin":true,"tty":true}],"restartPolicy":"Always","terminationGracePeriodSeconds":30,"dnsPolicy":"ClusterFirst","serviceAccountName":"default","serviceAccount":"default","nodeName":"x-x-x.18","securityContext":{}},"status":{"phase":"Running","conditions":[{"type":"Initialized","status":"True","lastProbeTime":null,"lastTransitionTime":"2016-07-20T08:59:59Z"},{"type":"Ready","status":"True","lastProbeTime":null,"lastTransitionTime":"2016-07-20T09:00:01Z"},{"type":"PodScheduled","status":"True","lastProbeTime":null,"lastTransitionTime":"2016-07-20T08:59:59Z"}],"hostIP":"x-x-x.18","podIP":"10.244.84.5","startTime":"2016-07-20T08:59:59Z","containerStatuses":[{"name":"lfec924f1f2c9cad","state":{"running":{"startedAt":"2016-07-20T09:00:00Z"}},"lastState":{},"ready":true,"restartCount":0,"image":"dhub.yunpro.cn/cuiweiqiang/crawler:latest","imageID":"docker://sha256:1d0a59110e4b438e5350830c4d59771fee82f57af1c43051e2eebe050953f930","containerID":"docker://8c1b4d29e2e0c90197351d2464fe6f29803590f95c1d997a4c7fa4d00777b46f"}]}}
I0720 17:59:09.671337   14663 round_trippers.go:299] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: kubectl/v1.3.2 (linux/amd64) kubernetes/9bafa34" http://localhost:8080/api
I0720 17:59:09.671738   14663 round_trippers.go:318] GET http://localhost:8080/api 200 OK in 0 milliseconds
I0720 17:59:09.671753   14663 round_trippers.go:324] Response Headers:
I0720 17:59:09.671757   14663 round_trippers.go:327]     Content-Type: application/json
I0720 17:59:09.671760   14663 round_trippers.go:327]     Date: Wed, 20 Jul 2016 09:59:09 GMT
I0720 17:59:09.671764   14663 round_trippers.go:327]     Content-Length: 131
I0720 17:59:09.671796   14663 request.go:891] Response Body: {"kind":"APIVersions","versions":["v1"],"serverAddressByClientCIDRs":[{"clientCIDR":"0.0.0.0/0","serverAddress":"x-x-x.6:443"}]}
I0720 17:59:09.671910   14663 round_trippers.go:299] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: kubectl/v1.3.2 (linux/amd64) kubernetes/9bafa34" http://localhost:8080/apis
I0720 17:59:09.672405   14663 round_trippers.go:318] GET http://localhost:8080/apis 200 OK in 0 milliseconds
I0720 17:59:09.672420   14663 round_trippers.go:324] Response Headers:
I0720 17:59:09.672424   14663 round_trippers.go:327]     Content-Type: application/json
I0720 17:59:09.672427   14663 round_trippers.go:327]     Date: Wed, 20 Jul 2016 09:59:09 GMT
I0720 17:59:09.672431   14663 round_trippers.go:327]     Content-Length: 1625
I0720 17:59:09.672463   14663 request.go:891] Response Body: {"kind":"APIGroupList","groups":[{"name":"apps","versions":[{"groupVersion":"apps/v1alpha1","version":"v1alpha1"}],"preferredVersion":{"groupVersion":"apps/v1alpha1","version":"v1alpha1"},"serverAddressByClientCIDRs":[{"clientCIDR":"0.0.0.0/0","serverAddress":"x-x-x.6:443"}]},{"name":"autoscaling","versions":[{"groupVersion":"autoscaling/v1","version":"v1"}],"preferredVersion":{"groupVersion":"autoscaling/v1","version":"v1"},"serverAddressByClientCIDRs":[{"clientCIDR":"0.0.0.0/0","serverAddress":"x-x-x.6:443"}]},{"name":"batch","versions":[{"groupVersion":"batch/v1","version":"v1"},{"groupVersion":"batch/v2alpha1","version":"v2alpha1"}],"preferredVersion":{"groupVersion":"batch/v1","version":"v1"},"serverAddressByClientCIDRs":[{"clientCIDR":"0.0.0.0/0","serverAddress":"x-x-x.6:443"}]},{"name":"extensions","versions":[{"groupVersion":"extensions/v1beta1","version":"v1beta1"}],"preferredVersion":{"groupVersion":"extensions/v1beta1","version":"v1beta1"},"serverAddressByClientCIDRs":[{"clientCIDR":"0.0.0.0/0","serverAddress":"x-x-x.6:443"}]},{"name":"policy","versions":[{"groupVersion":"policy/v1alpha1","version":"v1alpha1"}],"preferredVersion":{"groupVersion":"policy/v1alpha1","version":"v1alpha1"},"serverAddressByClientCIDRs":[{"clientCIDR":"0.0.0.0/0","serverAddress":"x-x-x.6:443"}]},{"name":"rbac.authorization.k8s.io","versions":[{"groupVersion":"rbac.authorization.k8s.io/v1alpha1","version":"v1alpha1"}],"preferredVersion":{"groupVersion":"rbac.authorization.k8s.io/v1alpha1","version":"v1alpha1"},"serverAddressByClientCIDRs":[{"clientCIDR":"0.0.0.0/0","serverAddress":"x-x-x.6:443"}]}]}
I0720 17:59:09.672849   14663 round_trippers.go:299] curl -k -v -XGET  -H "User-Agent: kubectl/v1.3.2 (linux/amd64) kubernetes/9bafa34" -H "Accept: application/json, */*" http://localhost:8080/api/v1/namespaces/cloud/pods/lfec924f1f2c9cad-4502j/log
I0720 17:59:09.675896   14663 round_trippers.go:318] GET http://localhost:8080/api/v1/namespaces/cloud/pods/lfec924f1f2c9cad-4502j/log 400 Bad Request in 3 milliseconds
I0720 17:59:09.675921   14663 round_trippers.go:324] Response Headers:
I0720 17:59:09.675941   14663 round_trippers.go:327]     Date: Wed, 20 Jul 2016 09:59:09 GMT
I0720 17:59:09.675948   14663 round_trippers.go:327]     Content-Length: 140
I0720 17:59:09.675953   14663 round_trippers.go:327]     Content-Type: application/json
I0720 17:59:09.676141   14663 helpers.go:173] server response object: [{
  "metadata": {},
  "status": "Failure",
  "message": "Unrecognized input header",
  "reason": "BadRequest",
  "code": 400
}]
F0720 17:59:09.676163   14663 helpers.go:108] Error from server: Unrecognized input header
```

**Comment by d23** [comment]
It seems to happen on all containers running with tty=true as soon as there is any output from them. 

```
$kubectl version
Client Version: version.Info{Major:"1", Minor:"3", GitVersion:"v1.3.3", GitCommit:"c6411395e09da356c608896d3d9725acab821418", GitTreeState:"clean", BuildDate:"2016-07-22T20:29:38Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"3", GitVersion:"v1.3.3", GitCommit:"c6411395e09da356c608896d3d9725acab821418", GitTreeState:"dirty", BuildDate:"1970-01-01T00:00:00Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"linux/amd64"}
```

```
$docker version
Client:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:20:08 2016
 OS/Arch:      darwin/amd64

Server:
 Version:      1.11.1
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   5604cbe
 Built:        Wed Apr 27 00:34:20 2016
 OS/Arch:      linux/amd64
```

**Comment by castrojo** [comment]
/assign

This seems to be an openshift specific issue so I'm going to close this.

**Comment by rm-you** [comment]
Are we sure this is an openshift only issue? I'm suffering from the same thing as @shenshouer and @d23 I think, specifically with `tty=true` and getting a 400 when trying to retrieve logs. They were both on kubernetes 1.3.x not openshift, right?

I am using:
```
Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.4", GitCommit:"d6f433224538d4f9ca2f7ae19b252e6fcb66a3ae", GitTreeState:"clean", BuildDate:"2017-05-19T20:41:07Z", GoVersion:"go1.8.1", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"3", GitVersion:"v1.3.5", GitCommit:"b0deb2eb8f4037421077f77cb163dbb4c0a2a9f5", GitTreeState:"clean", BuildDate:"2016-08-11T20:21:58Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"linux/amd64"}
```
Admittedly the kube cluster I'm on is a bit old.

```
I0726 12:29:28.455637   93945 round_trippers.go:398] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: kubectl/v1.6.4 (darwin/amd64) kubernetes/d6f4332" https://myapihost/api/v1/namespaces/myns/pods/mypod-4185495954-nau2g/log
I0726 12:29:28.720089   93945 round_trippers.go:417] GET https://myapihost/api/v1/namespaces/myns/pods/mypod-4185495954-nau2g/log 400 Bad Request in 264 milliseconds
I0726 12:29:28.720115   93945 round_trippers.go:423] Response Headers:
I0726 12:29:28.720119   93945 round_trippers.go:426]     Content-Type: application/json
I0726 12:29:28.720123   93945 round_trippers.go:426]     Content-Length: 140
I0726 12:29:28.720127   93945 round_trippers.go:426]     Date: Wed, 26 Jul 2017 19:29:28 GMT
I0726 12:29:28.720146   93945 request.go:991] Response Body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"Unrecognized input header","reason":"BadRequest","code":400}
I0726 12:29:28.721804   93945 helpers.go:206] server response object: [{
  "metadata": {},
  "status": "Failure",
  "message": "Unrecognized input header",
  "reason": "BadRequest",
  "code": 400
}]
F0726 12:29:28.721824   93945 helpers.go:119] Error from server (BadRequest): Unrecognized input header
```
This seems to line up with the two above, though I guess not the original report.
If I disable tty it works fine.

@ncdc mentioned a fix to `fsouza/go-dockerclient` but I'm not sure which fix he's referencing, and whether or not I'd have it.

If the two of you could let me know if the problems you were seeing were resolved in a newer version than 1.3.x, that'd be awesome -- there are a few interlinked threads on this that have people mentioning this specific issue, but none mention ever finding the cause or a fix, and eventually were closed or people switched focus. T_T

**Comment by rm-you** [comment]
Ok nevermind, I was able to verify myself, by getting access to a newer cluster:
```
Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.4", GitCommit:"d6f433224538d4f9ca2f7ae19b252e6fcb66a3ae", GitTreeState:"clean", BuildDate:"2017-05-19T20:41:07Z", GoVersion:"go1.8.1", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.0", GitCommit:"d3ada0119e776222f11ec7945e6d860061339aad", GitTreeState:"clean", BuildDate:"2017-06-29T22:55:19Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
```

It was definitely fixed there. So, anyone who runs into this -- I'm not sure exactly when it was fixed, but by 1.7 it's working.
